apps/frontend/src/app/5_pages/AavePage/components/BorrowPositionsList/components/EfficiencyModeCard/components/SwitchEModeForm/SwitchEModeForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  Button,
+  ErrorBadge,
+  ErrorLevel,
+  Icon,
+  IconNames,
+  Paragraph,
+  Select,
+  SimpleTable,
+  SimpleTableRow,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { config } from '../../../../../../../../../constants/aave';
+import { useAaveSetUserEMode } from '../../../../../../../../../hooks/aave/useAaveSetUserEMode';
+import { useAaveUserReservesData } from '../../../../../../../../../hooks/aave/useAaveUserReservesData';
+import { translations } from '../../../../../../../../../locales/i18n';
+import { EModeCategory } from '../../../../../../../../../types/aave';
+import { CollateralRatioHealthBar } from '../../../../../CollateralRatioHealthBar/CollateralRatioHealthBar';
+import { normalizeEModeSummary } from '../../EfficencyModeCard.utils';
+
+type SwitchEModeFormProps = {
+  current: EModeCategory;
+  categories: EModeCategory[];
+  onComplete: () => void;
+};
+
+export const SwitchEModeForm: FC<SwitchEModeFormProps> = ({
+  current,
+  categories,
+  onComplete,
+}) => {
+  const { handleSetUserEMode } = useAaveSetUserEMode();
+  const [category, setCategory] = useState<string>('');
+  const { summary, userReservesData, reservesData, timestamp } =
+    useAaveUserReservesData();
+
+  const categoriesOptions = useMemo(() => {
+    return categories
+      .filter(c => c.id !== current.id)
+      .map(category => ({
+        label: category.label,
+        value: String(category.id),
+      }));
+  }, [categories, current?.id]);
```

Comment:
filter + map = reduce

Reply:
Mmm sometimes I just think filter + map is cleaner. And for sure not a efficiency issue here as at most we'll have 3 elements... Anyways I did the update, just to open up the debate )